### PR TITLE
Make models/pool.py pylint compatible

### PR DESCRIPTION
**Pull request by j-y-matsubara**
---

This PR fixes some pylint errors and divide the airflow/ti_dep/dependencies.py to avoid cyclic import.

Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by j-y-matsubara** [comment]
Cyclic import message:
・Cyclic import (airflow.models -> airflow.models.pool -> airflow.ti_deps.deps.pool_slots_available_dep) (cyclic-import)
・Cyclic import (airflow.models -> airflow.models.baseoperator -> airflow.models.pool -> airflow.ti_deps.deps.pool_slots_available_dep) (cyclic-import)

File: airflow/ti_deps/deps/pool_slots_available_dep.py [hunk 1]
```diff
@@ -20,9 +20,7 @@
 
 from airflow.ti_deps.deps.base_ti_dep import BaseTIDep
 from airflow.utils.session import provide_session
-from airflow.utils.state import State
-
-STATES_TO_COUNT_AS_RUNNING = [State.RUNNING, State.QUEUED]
```
Comment (turbaszek) [inline airflow/ti_deps/deps/pool_slots_available_dep.py:25]: I like this change but it could be better to make this as separate PR, WDYT @mik-laj ?

**Comment by turbaszek** [comment]
Hi @j-y-matsubara ! Those changes look good but you've found the reason why those files are still on the todo list: cyclic imports. However, I think we should merge this change and not remove the file from the todo list. In this way, we will have less work in the future. WDYT @potiuk @kaxil ?

File: airflow/ti_deps/deps/pool_slots_available_dep.py (hunk 1 above)
Comment (mik-laj) [reply airflow/ti_deps/deps/pool_slots_available_dep.py:25]: In my opinion, it could be in this PR.

**Comment by j-y-matsubara** [comment]
Thank you for your comments, @turbaszek.
If should I cancel scripts/ci/pylint_todo.txt modifications, I will do it right away.

**Review by potiuk — APPROVED** [review]

**Comment by kaxil** [comment]
> Hi @j-y-matsubara ! Those changes look good but you've found the reason why those files are still on the todo list: cyclic imports. However, I think we should merge this change and not remove the file from the todo list. In this way, we will have less work in the future. WDYT @potiuk @kaxil ?

Yea that is fine 👍

**Comment by j-y-matsubara** [comment]
Thank you for your comments. 
I excluded scripts/ci/pylint_todo.txt from commit. ( not removing the file from the todo list. ) 
Is this alright with you, @turbaszek @kaxil @potiuk ?

**Comment by turbaszek** [comment]
> Thank you for your comments.
> I excluded scripts/ci/pylint_todo.txt from this commit. ( not removing the file from the todo list. )
> Is this alright with you, @turbaszek @kaxil @potiuk ?

Yeah! Can you fix the imports? It seems that isort is sad :<

**Comment by j-y-matsubara** [comment]
> > Thank you for your comments.
> > I excluded scripts/ci/pylint_todo.txt from this commit. ( not removing the file from the todo list. )
> > Is this alright with you, @turbaszek @kaxil @potiuk ?
> 
> Yeah! Can you fix the imports? It seems that isort is sad :<

Thank you for your comments, @turbaszek.
I fixed the imports.

**Comment by j-y-matsubara** [comment]
I don't know why this error is occurred in travis
```
tests/runtime/kubernetes/test_kubernetes_executor.py:203: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/runtime/kubernetes/test_kubernetes_executor.py:140: in ensure_dag_expected_state
    self.assertEqual(state, expected_final_state)
E   AssertionError: 'failed' != 'success'
E   - failed
E   + success
```
